### ISSUE TITLE
Create Spotify Integration

### ISSUE THREAD

**Issue by nickdibari** [Moody-Tunes/moodytunes#2]
~BLOCKED BY #1~ 
~BLOCKED BY #3~ 

BLOCKS #13 

Once the database is built, we can write a script that will pull songs from Spotify's API. This script will fetch a reasonable number of songs from Spotify, parse the response for the needed information, and create song records from the data. This should be written as a management command, to allow for access to Django's model library and settings configuration.

Previously we've gathered songs from playlists, I think we can use the same approach here. For a number of genres in a collection, fetch a batch of songs from a random playlist and save those. We should be sure to keep track of how many songs we create and limit to a configurable amount. Additionally, it might make sense to have some backoff between making requests to Spotify to avoid going over a rate limit.

The data we retrieve should be full: any songs without sentiment and valence should not be saved to the database. This can help to keep our data properly in order and avoid the issue of a nullable field.

Keys must be kept secure. Configure with environment variables or personal settings file.

Offer ability to run command from admin interface. We can use [django-admincommand](https://github.com/liberation/django-admincommand) to accomplish this.

Do we want to use a client? Or interface directly with API?

Deliverables
- Management command to create Song instances from Spotify API
- Audit trail of command execution
- Proper unit test coverage

**Comment by nickdibari** [comment]
Documentation
https://developer.spotify.com/documentation/web-api/

**Comment by nickdibari** [comment]
NOTE
We can't work on this until the database schema ticket is complete (we need tables to write to) plus I don't want to start working on any type of processing code until we get logging squared away. Ticket reflects the work needed to be done to unblock this ticket

**Comment by nickdibari** [comment]
NOTE
Let's double check the terms of service for Spotify's API when we finish this to see if we're pushing the boundary of what they allow users to do with their data. If it's too egregious we should find a way to scale it back, or contact them to see what we can do.

https://developer.spotify.com/terms/

**Comment by nickdibari** [comment]
NOTE
We're gonna include genre information with the song. Could offer some insight into what genres have what emotional context and what genres people find resonate with different emotions. Will add a column to the Song table to hold this information.

**Comment by nickdibari** [comment]
QUESTION

Because we have to look for the `genre` at the album level, we have to make one API call per track to get the genre information for the track. Should we refactor this to "spider" through a playlist instead? For a random few tracks, get the album for that track and store all of those tracks. Maybe we can just get a batch of albums from another endpoint, skipping the playlists thing entirely?

**Comment by nickdibari** [comment]
NOTE
Let's leave ourselves the opportunity for adding genre but do the work in a separate ticket. We don't need it for an MVP and can always add genre information retroactively.
